Add Fleet & Agent 8.13.0 Release Notes (#961)
elastic/ingest-docs: docs/en/ingest-management/release-notes/release-notes-8.13.asciidoc

:agent-pull: https://github.com/elastic/elastic-agent/pull/
:fleet-server-issue: https://github.com/elastic/fleet-server/issues/
:fleet-server-pull: https://github.com/elastic/fleet-server/pull/

[[release-notes]]
= Release notes

This section summarizes the changes in each release.

* <<release-notes-8.13.0>>

Also see:

* {kibana-ref}/release-notes.html[{kib} release notes]
* {beats-ref}/release-notes.html[{beats} release notes]

// begin 8.13.0 relnotes

[[release-notes-8.13.0]]
== {fleet} and {agent} 8.13.0

Review important information about {fleet-server} and {agent} for the 8.13.0 release.

[discrete]
[[security-updates-8.13.0]]
=== Security updates

{agent}::
* Update Go version to 1.21.8. {agent-pull}4221[#4221]

[discrete]
[[breaking-changes-8.13.0]]
=== Breaking changes

Breaking changes can prevent your application from optimal operation and
performance. Before you upgrade, review the breaking changes, then mitigate the
impact to your application.

// copied from Kibana release notes: https://github.com/elastic/kibana/pull/179216
[discrete]
[[breaking-176879]]
.Removes conditional topics for Kafka outputs
[%collapsible]
====

> *Details* +
> The Kafka output no longer supports conditional topics. For more information, refer to ({kibana-pull}176879[#176879]).

Reviewer: ```suggestion
*Details* +
The Kafka output no longer supports conditional topics while the final syntax is evaluated ahead of Kafka output GA. For more information, refer to ({kibana-pull}176879[#176879]).
```

I think we want to provide more context on this one, since there is no context in the linked issue.

//=== Known issues

[discrete]
[[new-features-8.13.0]]
=== New features

The 8.13.0 release added the following new and notable features.

{fleet}::
* Adds reference to `ecs@mappings` for each index template ({kibana-pull}174855[#174855]).
* Adds support for the `subobjects` setting on the object type mapping ({kibana-pull}171826[#171826]).

{fleet-server}::
* Add support for storing output secrets in a new `secrets` block. {fleet-server-pull}3061[3061] {fleet-server-issue}2966[2966]
* Add support for the remote {es} output type in {fleet-server}. {fleet-server-pull}3051[3051]
* Report the health state of remote {es} outputs to the `logs-fleet_server.output_health-default` data stream. {fleet-server-pull}3127[3127] {fleet-server-issue}3116[3116]
* Add a `policy_debounce_time` configuration to add a forced delay to the policy index monitor when it successfully gathers new documents. {fleet-server-pull}3234[3234]

{agent}::
* Log a summary of each policy configuration change received from {fleet}. {agent-pull}4050[#4050] {agent-issue}3406[#3406]
* Add the full version number to the installation directory name. {agent-pull}4193[#4193] {agent-issue}2579[#2579]

> * Make more selective the Pod autodiscovery upon node and namespace update events. {agent-pull}4226[#4226] {beats-issue}37338[#37338]

Reviewer: ```suggestion
* Ignore Kubernetes node and namespace update events that do not change pod metadata. {agent-pull}4226[#4226] {beats-issue}37338[#37338]
```

* Create the `.installed` marker earlier on in the install process, allowing the use of `elastic-agent uninstall` to cleanup if the install fails. {agent-pull}4172[#4172] {agent-issue}4051[#4051]
* Add a postrm script to {agent} DEB and RPM packages. {agent-pull}4334[#4334] {agent-issue}3784[#3784] {agent-issue}4267[#4267]
* Kubernetes secrets provider has been improved to update a Kubernetes secret  when the secret value changes. {agent-pull}4371[#4371] {agent-issue}4168[#4168]
* Upgrade link:https://github.com/elastic/elastic-agent-system-metrics[elastic-agent-system-metrics] to version 0.9.2. {agent-pull}4383[#4383]

[discrete]
[[enhancements-8.13.0]]
=== Enhancements

{fleet}::
* Adds `skipRateLimitCheck` flag to the Upgrade API and Bulk_upgrade API ({kibana-pull}176923[#176923]).
* Adds making datastream rollover lazy ({kibana-pull}176565[#176565]).
* Stops creating  the `{type}-{datastet}@custom` component template during package installation ({kibana-pull}175469[#175469]).
* Adds the `xpack.fleet.isAirGapped` flag ({kibana-pull}174214[#174214]).

> * Adds a warning when download upgrade is failing ({kibana-pull}173844[#173844]).

Reviewer: ```suggestion
* Add a warning when downloading the new version in an agent upgrade is failing ({kibana-pull}173844[#173844]).
```

* Fixes asset creation during custom integration installation ({kibana-pull}174869[#174869]).
* Fixes cascading agent policy's namespace to package policies ({kibana-pull}174776[#174776]).

{fleet-server}::
* Add missing `Elastic-Api-Version` and `X-Request-Id` headers to the {fleet-server} OpenAPI specification. {fleet-server-pull}3044[3044]
* Replace all secret references in input objects. {fleet-server-pull}3086[3086] {fleet-server-issue}3083[3083]
* Deprecate the redundant `fleet.agent.logging.level` attribute. {fleet-server-pull}3195[3195] {fleet-server-issue}3126[3126]
* Add validation to make sure that status and message are present in the checkin API request body.  {fleet-server-pull}3233[3233] {fleet-server-issue}2420[2420]
* Fix a bug where agents were stuck in non-upgradeable state after an upgrade. {fleet-server-pull}3264[3264] {fleet-server-issue}3263[3263]
* Fix chunked file delivery so that files are delivered in order. {fleet-server-pull}3283[#3283]
* Fix a bug where the self monitor stops output health reporting if the output configuration is not acknowledged by agents. {fleet-server-pull}3335[#3335] {fleet-server-issue}3334[3334]

{agent}::

> * Fix component control protocol to allow checkin to be chunked across multiple messages. {agent-pull}3884[#3884] {agent-issue}2460[#2460]

Reviewer: ```suggestion
* Fix component control protocol to allow checkin to be chunked across multiple messages. Fixes errors related to the gRPC max message size being exceeded. {agent-pull}3884[#3884] {agent-issue}2460[#2460]
```
